impexp-plugin-api/src/main/java/org/citydb/plugin/cli/CliOptionBuilder.java

```diff
@@ -0,0 +1,309 @@
+                } catch (NumberFormatException e) {
+                    throw new CommandLine.ParameterException(commandLine,
+                            "Error: The coordinates of a bounding box must be floating point numbers but were '" +
+                                    String.join(",", parts[0], parts[1], parts[2], parts[3]) + "'");
+                }
+
+                if (parts.length == 5) {
+                    try {
+                        boundingBox.setSrs(Integer.parseInt(parts[4]));
+                    } catch (NumberFormatException e) {
+                        throw new CommandLine.ParameterException(commandLine,
+                                "Error: The SRID of a bounding box must be an integer but was '" + parts[4] + "'");
+                    }
+                }
+
+                return boundingBox;
+            } else {
+                throw new CommandLine.ParameterException(commandLine,
+                        "A bounding box must be in MINX,MINY,MAXX,MAXY[,SRID] format but was '" + bbox + "'");
+            }
+        }
+
+        return null;
+    }
+
+    public static NamespaceContext namespaceContext(Map<String, String> namespaces) {
+        CityGMLNamespaceContext namespaceContext = new CityGMLNamespaceContext();
+        Modules.getADEModules().forEach(m -> namespaceContext.setPrefix(m.getNamespacePrefix(), m.getNamespaceURI()));
+        namespaceContext.setPrefixes(CityGMLVersion.v2_0_0);
+
+        if (namespaces != null) {
+            namespaces.forEach(namespaceContext::setPrefix);
+        }
+
+        return namespaceContext;
+    }
+
+    public static FeatureTypeFilter featureTypeFilter(String[] typeNames, NamespaceContext namespaceContext, CommandLine commandLine) {
+        if (typeNames != null) {
+            FeatureTypeFilter featureTypeFilter = new FeatureTypeFilter();
+            for (String typeName : typeNames) {
+                String[] parts = typeName.split(":");
+                if (parts.length == 2) {
+                    String namespace = namespaceContext.getNamespaceURI(parts[0]);
+                    if (namespace.equals(XMLConstants.NULL_NS_URI)) {
+                        throw new CommandLine.ParameterException(commandLine, "Unknown prefix: " + parts[0]);
+                    }
+
+                    Module module = Modules.getModule(namespace);
+                    if (module == null || !module.hasFeature(parts[1])) {
+                        throw new CommandLine.ParameterException(commandLine, "Unknown type name: " + typeName);
+                    }
+
+                    featureTypeFilter.addTypeName(new QName(namespace, parts[1]));
+                } else if (parts.length == 1) {
+                    Module module = Stream.concat(CityGMLVersion.v2_0_0.getCityGMLModules().stream(),
+                            Modules.getADEModules().stream())
+                            .filter(m -> m.hasFeature(parts[0]))
+                            .findFirst()
```

Comment:
My initial rationale was to avoid finding the same feature type name in two ADE modules where the first one is associated with CityGML 2.0 and the second one with CityGML 1.0. Mixing feature types from different CityGML versions in one feature type filter is not supported, and this was meant as a fail-fast check for such situations.

But you are right, the check is too restrictive because it excludes valid situations. Fixed with 909b4bf4a333e593104c5e1b2b65e405d7d1f04a